astronomer/providers/google/cloud/triggers/bigquery.py

```diff
@@ -378,3 +383,83 @@ async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
+        self.project_id = project_id
+        self.table_id = table_id
+        self.google_cloud_conn_id: str = google_cloud_conn_id
+        self.poll_interval = poll_interval
+        self.hook_params = hook_params
+
+    def serialize(self) -> Tuple[str, Dict[str, Any]]:
+        """Serializes BigQueryTableExistenceTrigger arguments and classpath."""
+        return (
+            "astronomer.providers.google.cloud.triggers.bigquery.BigQueryTableExistenceTrigger",
+            {
+                "dataset_id": self.dataset_id,
+                "project_id": self.project_id,
+                "table_id": self.table_id,
+                "google_cloud_conn_id": self.google_cloud_conn_id,
+                "poll_interval": self.poll_interval,
+                "hook_params": self.hook_params,
+            },
+        )
+
+    def _get_async_hook(self) -> BigQueryTableHookAsync:
+        return BigQueryTableHookAsync(gcp_conn_id=self.google_cloud_conn_id)
+
+    async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
+        """Will run until the table exists in the Google Big Query."""
+        while True:
+            try:
+                hook = self._get_async_hook()
+                response = await self._table_exists(
+                    hook=hook, dataset=self.dataset_id, table_id=self.table_id, project_id=self.project_id
+                )
+                if response:
+                    yield TriggerEvent({"status": "success", "message": "success"})
+                    return
+                await asyncio.sleep(self.poll_interval)
+            except Exception as e:
+                self.log.exception("Exception occurred while checking for Table existence")
+                yield TriggerEvent({"status": "error", "message": str(e)})
+                return
+
+    async def _table_exists(
+        self, hook: BigQueryTableHookAsync, dataset: str, table_id: str, project_id: str
+    ) -> bool:
+        """
+        Create client session and make call to BigQueryTableHookAsync and check for the table in Google Big Query.
+
+        :param hook: BigQueryTableHookAsync Hook class
+        :param dataset:  The name of the dataset in which to look for the table storage bucket.
+        :param table_id: The name of the table to check the existence of.
+        :param project_id: The Google cloud project in which to look for the table.
+            The connection supplied to the hook must provide
+            access to the specified project.
+        """
+        async with ClientSession() as session:
+            try:
+                client = await hook.get_table_client(
+                    dataset=dataset, table_id=table_id, project_id=project_id, session=session
+                )
+                response = await client.get()
+                return True if response else False
+            except ClientResponseError as err:
+                if err.status == 404:
+                    return False
```

Comment:
In the case of 404 i.e not found why we are not returning error ?

Reply:
GET https://bigquery.googleapis.com/bigquery/v2/projects/{projectId}/datasets/{datasetId}/tables/{tableId}

In this case it makes the API call to the google big query,  if the table exists it returns the table if not it returns 404 error in the trigger so we are polling to the same url till the table is found